README.md

```diff
@@ -98,4 +98,466 @@ On the joining screen, choose to join the meeting without `CallKit` or join via
+
+```swift
+class MyVideoTileObserver: VideoTileObserver { 
+    func videoTileDidAdd(tileState: VideoTileState) {
+        if (tileState.isContent) {
+            // tileState.attendeeId is formatted as "attendee-id#content"
+            let attendeeId = tileState.attendeeId
+            // Get the attendee ID from "attendee-id#content"
+            let baseAttendeeId = DefaultModality(attendeeId).base()
+            logger.info(msg: "$baseAttendeeId is sharing screen")
+
+            let screenVideoView = /* a VideoRenderView object in your application to show the video */
+            meetingSession.audioVideo.bindVideoView(videoView: screenVideoView, tileId: tileState.tileId)
+        }
+    }
+
+    func videoTileDidRemove(tileState: VideoTileState) {
+        meetingSession.audioVideo.unbindVideoView(tileId: tileState.tileId)
+    }
+    
+    meetingSession.audioVideo.addVideoTileObserver(observer: self)
+}
+```
+
+### Metrics
+
+#### Use case 20. Add an observer to receive the meeting metrics.
+
+See `ObservableMetric` for more available metrics and to monitor audio, video, and content share quality.
+
+```swift
+class MyMetricsObserver: MetricsObserver {
+    func metricsDidReceive(metrics: [AnyHashable: Any]) {
+        logger.info(msg: "Media metrics have been received: \(metrics)")
+    }
+    
+    meetingSession.audioVideo.addMetricsObserver(observer: self)
+}
+```
+
+### Data Message
+
+#### Use case 21. Add an observer to receive data message.
+
+You can receive real-time messages from multiple topics after starting the meeting session.
+
+```swift
+class MyDataMessageObserver: DataMessageObserver {
+    let topic = "chat"
+    // A throttled message is returned by backend from local sender
+    func dataMessageDidReceived(dataMessage: DataMessage) {
+        logger.info(msg: "\(dataMessage.timestampMs) \(dataMessage.text()) \(dataMessage.senderAttendeeId)")
+    }
+    
+    // You can also subscribe to multiple topics.
+    meetingSession.audioVideo.addRealtimeDataMessageObserver(topic: topic, observer: self)
+}
+```
+
+#### Use case 22. Send data message.
+
+You can send real time message to any topic, to which the observers that have subscribed will be notified.
+
+> Note: Topic needs to be alpha-numeric and it can include hyphen and underscores. Data cannot exceed 2kb and lifetime is optional but positive integer.
+
+```swift
+let topic = "chat"
+
+do {
+    // Send "Hello Chime" to any subscribers who are listening to "chat" topic with 1 seconds of lifetime
+    try meetingSession
+        .audioVideo
+        .realtimeSendDataMessage(topic: topic,
+                                data: "Hello Chime",
+                                lifetimeMs: 1000)
+} catch {
+    logger.error(msg: "Failed to send message!")
```

Comment:
It throws different exception for root cause: https://github.com/aws/amazon-chime-sdk-ios/blob/master/AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift#L506

Reply:
Sorry I read the wrong function. Updated